### DISCUSSION TITLE
preview composable functions

### DISCUSSION THREAD

**Discussion by javad0470** [varabyte/kobweb#348] [Q&A]
I am reaching out to inquire whether it is feasible to preview composed code without the need for running a server. Can we, for instance, view compose codes directly on the browser like HTML and CSS? The reason for this query stems from the considerable strain that the Kobweb server places on our system, depleting a significant amount of energy and putting undue pressure on the system's resources.

Your prompt attention to this matter would be greatly appreciated.

Thank you.

**Comment by bitspittle** [comment]
Hey, as mostly a solo dev working on this thing, I'm afraid it will be a long time before I could prioritize anything like in-IDE previews. This is a VERY difficult feature request.  The best way forward for a while really is to just run `kobweb run` locally on your machine.

I'm not sure what you mean by "considerable strain", maybe you can be more specific about how you diagnosed that? What version of Kobweb are you using? What OS are you using?

The Kobweb server is fundamentally a fairly straightforward Ktor web server. I have had lots of people running it so far across multiple kinds of machines, but this is the first I've heard of anyone mentioning undo strain. I am sure the JetBrains IDE, or programs like web browsers, are many many many many times more expensive to run than a simple local web server (unless there's a bug of some sort? Do you have a repro case?)
